src/components/Settings/Settings-devices.tsx

```diff
@@ -29,10 +29,20 @@ export class SettingsDevices extends Component<Props, State> {
 
     componentDidMount() {
         this.removeRedundantPopups();
+        this.focusOnToggle();
+    }
+    componentDidUpdate() {
+        this.focusOnToggle();
+    }
+
+    focusOnToggle = () => {
         if (this.menuRef && this.menuRef.children[0]) {
-            (this.menuRef.children[0] as HTMLElement).focus();
+            const toggleElement = (this.menuRef.children[0] as HTMLElement).getElementsByTagName(
+                "input"
+            );
+            toggleElement.length > 0 && toggleElement[0].focus();
         }
-    }
+    };
```

Comment:
the calling function should not know what focusOnToggle is doing. doing validation every time is code duplocation and lack of modularuty

Reply:
calling `this.focusOnToggle()` should toggle without any internal conditions.
if you wish to have another check function, I'm fine with `_menuExists()` for example.
and then you can call it from outside:
```
_menuExists = () => {
  const { menuRef } = this;
  if(!menuRef || !menuRef.children.length){
    return false;
  }
  return true;
}

componentDidUpdate() {
  if(!this._menuExists()){
    return;
  }
  this.focusOnToggle();
}
```

Reply:
@tomm1990  when we have a function  that handle a logic, it should contain also it's constraints (most of the time).
The idea is that if other will use this function - we will not need to repeat the if exist.

I think that Channa did it fine.

Reply:
The word toggle means to toggle. It's an action.
As a result, when you call that function, it should be fired without any internal conditions.
As this function contains internal conditions, calling it will not always toggle.
It may be confusing for a developer to understand why he fired this function and nothing happened.

suggested solutions:
1. move the conditions to the outside of this function.
2. rename that function to `focusOnToggleIfMenuExists` or `focusOnToggleIfPossible`.

Reply:
renamed